cmake/kokkos_arch.cmake

```
@@ -135,6 +135,7 @@ endif()
 
 kokkos_arch_option(INTEL_GEN GPU "SPIR64-based devices, e.g. Intel GPUs, using JIT" "KOKKOS_SHOW_SYCL_ARCHS")
 kokkos_arch_option(INTEL_DG1 GPU "Intel Iris XeMAX GPU" "KOKKOS_SHOW_SYCL_ARCHS")
+kokkos_arch_option(INTEL_DG2 GPU "Intel Arc Alchemist Xe-HPG GPU" "KOKKOS_SHOW_SYCL_ARCHS")
```

Intel® Flex 170 and  Intel® Flex 140 would also use this flag.

---

@masterleinad thanks for the QC. I just dug through the intel-llvm-sycl compiler and found out the correct device value `acm-g10` - I'm working with an intel arc A750

---

do you think the architecture flag should better reflect the sycl argument or stick with this value? I can make those changes today.

---

I would be curious if you see any difference in terms of compilation time. I don't think the `dg2` flag implies compiling for multiple architectures but it would be good to check.
If there is no difference, I think the current flag is good enough allwoing some flexibility.

---

Since we are going with `dg2` now, I think we should also document that instead of `Alchemist` in particular (but examples of GPUs would go into the documentation).
```suggestion
kokkos_arch_option(INTEL_DG2 GPU "Intel DG2 GPU" "KOKKOS_SHOW_SYCL_ARCHS")
```

---

complete!

---

Thanks!